.eslintrc.js

```diff
@@ -10,17 +10,6 @@ module.exports = {
   },
   plugins: ["filenames"],
   rules: {
-    "prettier/prettier": [
```

Comment:
Why are we removing this ? It actually prettifies the code while saving, maintains consistencies otherwise if you have different settings in your editor, it will take priority.

Reply:
as We have extended from `airbnb-typescript-prettier`, and extended with `prettier` [plugin](https://github.com/toshi-toma/eslint-config-airbnb-typescript-prettier/blob/f92ba46dc3f401dfa5c8debd0403daca3d9d9ece/index.js#L15) as well. This rules is already defined once  there. No need to mention it again in our file. 

https://github.com/toshi-toma/eslint-config-airbnb-typescript-prettier/blob/f92ba46dc3f401dfa5c8debd0403daca3d9d9ece/index.js#L19

Reply:
Yes some of them. But specifically here, singlequote, printwidth, printwidth etc.

Reply:
https://github.com/prettier/eslint-plugin-prettier#options, based on doc, the rules mentioned here will override the values from .prettierrc file. As those are infact default values of .prettierrc, that's why we don't need to mention them again.

Reply:
Ah got it, yes we can remove the prettierrc file since we have the rules here. We had to add this here so it becomes part of eslint fix.

Reply:
we shouldn't remove the prettierc file, it's being read by the prettier package installed in package.json and it's better to separate the cosmetic error and lint error. Note from the eslint prettier plugin 

> Note: While it is possible to pass options to Prettier via your ESLint configuration file, it is not recommended because editor extensions such as prettier-atom and prettier-vscode will read [.prettierrc](https://prettier.io/docs/en/configuration.html), but won't read settings from ESLint, which can lead to an inconsistent experience.

Reply:
Comes back to the point, im guessing why we added here was because airbnb-typescript-prettier uses certain rules and we have `"source.fixAll.eslint": true` on save action it conflicts the prettierrc rules, maybe we should think how we can still keep the following rules and have it only in one place.

Reply:
airbnb-typescript-prettier is using [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) plugin which turns off all the eslint rules that conflicts with Prettier, After that it uses [eslint-prettier-plugin](https://github.com/prettier/eslint-plugin-prettier) which uses Prettier Module installed as Dependency ( along with .prettierrc). If we run the command in the branch
`yarn eslint "src/**/*.{ts,tsx}"`, we won't find any eslint error because the options mentioned in the `.eslintrc.js` file  previously is not overriding anything. And if you modify the `.prettierrc.js` (e.g `singleQuote : true`), and run the command you will get eslint errors. That means, `prettierrc.js` is the only file where we should put all formatting related rules so that IDEs and the eslint plugin both can use the same file to format and find the error.

Reply:
Thanks @atiqmlbd  for looking it up. Lt time i checked we were seeing issues with vscode formatOnSave, will test it again and see if this is not the case now and we remove it from eslint.

Reply:
sometimes Vscode needs to be restarted to keep in sync with `prettierrc.js ` once configuration is changed